src/networking/nc/mod.rs

```diff
@@ -0,0 +1,958 @@
+    Err(MesaError::from(
+        failure::err_msg(format!("{}", err_msg)).compat()
+        ))
+}
+
+fn build_ports(ports: &str) -> Result<Vec<u16>, MesaError>{
+    // TODO: suport XX-XX
+    let port_list = match ports.parse::<u16>() {
+        Ok(port) => port,
+        Err(_) => {
+            return mesaerr_result(&format!("invalid port[s] {}", ports));
+        }
+    };
+
+    Ok(vec!(port_list))
+}
+
+fn warn(msg: &str) {
+    eprint!("{}", msg);
+}
+
+fn debug_info(msg: &str) {
+    if PRINT_DEBUG_INFO {
+        eprint!("{}", msg);
+    }
+}
+
+impl NcOptions {
+    pub fn parse(matches: ArgMatches, msg: &str) -> Result<NcOptions, MesaError> {
+        let mut portlist = vec!();
+        let lflag = matches.is_present("l");
+        let mut host = String::from("127.0.0.1");
+        let uport:String;
+        let mut interval = None;
+        let mut timeout = None;
+        let s_addr = match matches.value_of("s") {
+            Some(addr) => Some(String::from(addr)),
+            None => None
+        };
+        let uflag = matches.is_present("u");
+        let pflag = matches.is_present("p");
+
+        let zflag = matches.is_present("z");
+        let kflag = matches.is_present("k");
+
+        /* Cruft to make sure options are clean, and used properly. */
+        let positionals: Vec<&str> = if matches.is_present("positionals") {
+            matches.values_of("positionals").unwrap().collect()
+        } else {
+            vec!()
+        };
+
+        let family = if matches.is_present("U") {
+            AF_UNIX
+        } else if matches.is_present("6") {
+            AF_INET6
+        } else if matches.is_present("4") {
+            AF_INET
+        } else {
+            AF_UNSPEC
+        };
+
+        if positionals.len() == 1 {
+            if family == AF_UNIX {
+                host = String::from(positionals[0]);
+                uport = String::new();
+            } else {
+                if !lflag {
+                    return mesaerr_result(msg);
+                }
+                uport = String::from(positionals[0]);
+            }
+        } else if positionals.len() >= 2 {
+            host = String::from(positionals[0]);
+            uport = String::from(positionals[1]);
+        } else {
+            return mesaerr_result(msg);
+        }
+
+        if lflag && s_addr.is_some() {
+            return mesaerr_result("cannot use -s and -l");
+        }
+        if lflag && pflag {
+            return mesaerr_result("cannot use -p and -l");
+        }
+        if lflag && zflag {
+            return mesaerr_result("cannot use -z and -l");
+        }
+        if !lflag && kflag {
+            return mesaerr_result("must use -l with -k");
+        }
+
+        if !uport.is_empty() {
+            portlist = build_ports(&uport)?;
+        }
+
+        if matches.is_present("i") {
+            let sec = matches.value_of("i").unwrap().parse::<u64>().unwrap();
```

Comment:
You can verify that this is valid using a [validator](https://kbknapp.github.io/clap-rs/clap/struct.Arg.html#method.validator).